Sources/Plasma/FeatureLib/pfPasswordStore/pfPasswordStore.cpp

```diff
@@ -49,21 +51,22 @@ You can contact Cyan Worlds, Inc. by email legal@cyan.com
 /* Get the pfPasswordStore instance */
 pfPasswordStore* pfPasswordStore::Instance()
 {
-    static pfPasswordStore* store = nullptr;
+    static std::unique_ptr<pfPasswordStore> store;
 
-    if (store == nullptr) {
+    if (!store) {
 #if defined(HS_BUILD_FOR_WIN32)
-        store = new pfWin32PasswordStore();
+
+        store = std::make_unique<pfWin32PasswordStore>();
 #elif defined(HS_BUILD_FOR_OSX)
-        store = new pfMacPasswordStore();
+        store = std::make_unique<pfMacPasswordStore>();
 #elif defined(HAVE_LIBSECRET)
-        store = new pfUnixPasswordStore();
+        store = std::make_unique<pfUnixPasswordStore>();
 #else
-        store = new pfFilePasswordStore();
+        store = std::make_unique<pfFilePasswordStore>();
 #endif
     }
 
-    return store;
+    return store.get();
 }
```

Comment:
This whole function could be done without dynamic allocation...

```c++
pfPasswordStore* pfPasswordStore::Instance()
{
#if defined(HS_BUILD_FOR_WIN32)
    static pfWin32PasswordStore store;
#elif defined(HS_BUILD_FOR_OSX)
    static pfMacPasswordStore store;
#elif defined(HAVE_LIBSECRET)
    static pfUnixPasswordStore store;
#else
    static pfFilePasswordStore store;
#endif

    return &store;
}
```

Now that we depend on MSVC2015, this is guaranteed to be thread safe.

Reply:
Good point!